### PR TITLE
Move admin flag from person to local_user (fixes #3060)

### LINKED ISSUE #3060: Admin was removed either by bug or exploit.
I am admin of lemmy.world.
Suddenly I was missing the 'admin' icon in the top menu.
I checked, I also wasn't listed on the home page as 'admin'.

I then checked the database, the 'admin' column was 'f' for my account.
There's nothing in the modlog about it, and nothing in the mod_add table.
The 3 other admins are trusted and they are asleep (US timezone).

I have fixed it by updating the person table. But this could be a serious bug or security exploit.

### PR DESCRIPTION
**Pull request by Nutomic**
The person table is for federated data, but admin flag can only apply to local users. Thats why it really belongs in the local_user table. This will also prevent the federation code from accidentally overwriting the admin flag

### CODE REVIEW

**Comment by Nutomic** [comment]
This is not a breaking change, the API is unchanged.

File: crates/db_schema/src/source/local_user.rs [hunk 1]
```diff
@@ -51,6 +51,8 @@ pub struct LocalUser {
   pub totp_2fa_secret: Option<String>,
   /// A URL to add their 2-factor auth.
   pub totp_2fa_url: Option<String>,
+  /// Whether the person is an admin.
+  pub admin: bool,
```
[the remark this answers is not in the document]
Comment (Nutomic) [reply crates/db_schema/src/source/local_user.rs:55]: Right this is actually included in the API

File: crates/api/src/local_user/add_admin.rs [hunk 2]
```diff
@@ -27,20 +28,21 @@ impl Perform for AddAdmin {
     // Make sure user is an admin
     is_admin(&local_user_view)?;
 
-    let added = data.added;
-    let added_person_id = data.person_id;
-    let added_admin = Person::update(
+    let target = LocalUserView::read_person(context.pool(), data.person_id).await?;
```
[the remark this answers is not in the document]
Comment (dessalines) [reply crates/api/src/local_user/add_admin.rs:31]: Yeah, might as well do this change across the board. Might mean fewer DB calls.

**Comment by Nutomic** [comment]
Resolved conflicts. For some reason api tests are failing with this error, no idea why:
```

$ jest -i follow.spec.ts && jest -i src/post.spec.ts && jest -i comment.spec.ts && jest -i private_message.spec.ts && jest -i user.spec.ts && jest -i community.spec.ts
FAIL  src/follow.spec.ts
✕ Follow federated community

● Follow federated community

thrown: "unknown"

10 | } from "./shared";
11 |
> 12 | beforeAll(async () => {
| ^
13 |   await setupLogins();
14 | });
15 |

at Object.<anonymous> (src/follow.spec.ts:12:1)
```

**Comment by dessalines** [comment]
I'll test it out now

**Comment by dessalines** [comment]
The admin column on the source table was in the wrong order. It needs to match the order in schema.rs. This was hard to find because they were both boolean columns.

**Review by dessalines — APPROVED** [review]

**Comment by dessalines** [comment]
This [test](https://woodpecker.join-lemmy.org/repos/129/pipeline/1796/18) started failing all the sudden.

[Here's the failing line](https://github.com/LemmyNet/lemmy/blob/main/api_tests/src/post.spec.ts#L391)

I don't understand why its suddenly unable to resolve a banned person. Maybe because federated bans don't work on admins?

**Pull request merged**